### ISSUE TITLE
File sizes appear larger in visualizer than in reality

### ISSUE THREAD

**Issue by toBeOfUse** [KusStar/vite-bundle-visualizer#8]
The file sizes shown in the visualizer are larger than the ones reported by Vite on the command line or that show up on disk after building. To see this, you can just clone [vitesse](https://github.com/antfu/vitesse), install its packages, and run vite-bundle-visualizer for it.

On the command line, a file size of 153kb is shown for the main javascript asset:

![image](https://github.com/KusStar/vite-bundle-visualizer/assets/49729978/4fb7771e-cc55-41c7-aa29-c8e2a5bed328)

Similarly, on disk, the file appears to be 150kb:

![image](https://github.com/KusStar/vite-bundle-visualizer/assets/49729978/f6da2884-e07f-4f8f-985c-dfd3c3335cf5)

But in the visualizer, the same file (with the same hash in the name, so it should have the same contents) is reported to be 461kb:

![image](https://github.com/KusStar/vite-bundle-visualizer/assets/49729978/67bc5125-8399-477d-8887-661975276ca1)

**Comment by KusStar** [comment]
It seems like only happen in [vitesse](https://github.com/antfu/vitesse) or `vite-ssg` project? Looking weird, I will try to figure it out.

**Comment by KusStar** [comment]
[v0.10.0](https://www.npmjs.com/package/vite-bundle-visualizer/v/0.10.0) involves `--sourcemap` option, it will more accurate but result less detailed.

![image](https://github.com/KusStar/vite-bundle-visualizer/assets/21271495/592bcd09-1f83-41dd-8eb4-9098394073f6)

**Comment by samosad** [comment]
same problem for me!

real bundle size:
<img width="1080" alt="image" src="https://github.com/KusStar/vite-bundle-visualizer/assets/1986478/1f76582c-19a8-41b4-b31c-6896cd928e46">

same bundle in visualizer:
<img width="841" alt="image" src="https://github.com/KusStar/vite-bundle-visualizer/assets/1986478/ca9e48a9-f505-4ccb-ba73-32f2dc53113b">

with `--sourcemap` option:
<img width="835" alt="image" src="https://github.com/KusStar/vite-bundle-visualizer/assets/1986478/65f2d6a0-a279-4ffc-808e-4f2aba80e207">

**Comment by bfricka** [comment]
The sourcemap option is indeed less detailed, and also not more accurate, in my experience. E.g. I separated out a lodash chunk to test it, and the build (and file system) show that chunk as 34.18 kB, whereas the sourcemap options shows it as 185.66 KB (which is 190.12 kB), so it's off by 5.5x. Another chunk, however measured 2.2x off.

This hurts the functional usefulness of the tool dramatically because the relative sizes of modules and chunks are not accurate. It wouldn't be such an issue if just the numbers were inaccurate, but with the relative sizing being so dramatically off, this tool is really only useful as a module graph.

Without the sourcemap option, the numbers appear identical, but there is gzip info at least.

Edit: Appears to match [this issue](https://github.com/btd/rollup-plugin-visualizer/issues/163). Since it's downstream, and I don't see anything wrong with this Vite wrapper code, there's probably not much the author can do. This plugin is already using the latest, and it doesn't appear to help.

**Comment by KusStar** [comment]
> The sourcemap option is indeed less detailed, and also not more accurate, in my experience. E.g. I separated out a lodash chunk to test it, and the build (and file system) show that chunk as 34.18 kB, whereas the sourcemap options shows it as 185.66 KB (which is 190.12 kB), so it's off by 5.5x. Another chunk, however measured 2.2x off.
> 
> This hurts the functional usefulness of the tool dramatically because the relative sizes of modules and chunks are not accurate. It wouldn't be such an issue if just the numbers were inaccurate, but with the relative sizing being so dramatically off, this tool is really only useful as a module graph.
> 
> Without the sourcemap option, the numbers appear identical, but there is gzip info at least.
> 
> Edit: Appears to match [this issue](https://github.com/btd/rollup-plugin-visualizer/issues/163). Since it's downstream, and I don't see anything wrong with this Vite wrapper code, there's probably not much the author can do. This plugin is already using the latest, and it doesn't appear to help.

Yeah, it seems the output bundles information of Vite(internal is rollup) is inaccurate, you only can get accurate final size of a module bundle, which are tree-shaken results, but only get size of a module before tree-shaken in module graph. Result in inaccurate in some cases.

**Comment by stefan-schweiger** [comment]
@KusStar seems like this issue was fixed in rollup-plugin-visualizer, would you mind updating your package to the latest version?
https://github.com/btd/rollup-plugin-visualizer/issues/163

**Comment by stefan-schweiger** [comment]
Oh sorry, just saw that you are already on this version :(

**Comment by samesfahani-tuplehealth** [comment]
Any workarounds for this issue? I'm seeing the same issue. When I visualize deps, it seems like it's not honoring `build.rollupOptions.treeshake` where was `vite build` does, rendering the actual usefulness of the visualization useless.

**Comment by gavrashenko** [comment]
Are there any updates? It's too hard to analyze deps if I can't trust the data (

**Comment by SimonEast** [comment]
I've just noticed this problem too. It's not very useful if the sizes are all off.

**Comment by HummingMind** [comment]
Same issue here.
"vite-bundle-analyzer" plugin does not have this issue.

Seems like the problem is within the underlying rollup-plugin-visualizer package. Probably nothing that can be done to fix it in vite-bundle-visualizer.

Thank you! 🍻
